### PR TITLE
App UI

### PR DESCRIPTION
**Pull request by dev89s**
Added a file named app.rb which includes a class named `App`.
It begins the app by calling begin function in the `main` method in the app.rb file
The options to select in the UI:
  1. List all books
  2. List all people
  3. Create a person (Asks whether to create a `student` or a `teacher`) and asks related details
  4. Create a book
  5. Create a rental by selecting already created persons and books to add to the rental record
  6. List all rentals for a given person id
The user will be presented the UI as a list of options that could be selected to enter other menus to do the required task.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+      list_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rentals_by_id
+    end
```
Comment (christianbiring1) [inline app.rb]: - [ ] Kindly make sure you have the `7th` options to exit the app as you may notice you did not implement it.
```rb
when '7'
# The message you may want to write :)
exit
end
```

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  def operate(command)
+    case command
+    when '1'
+      list_books
```
Comment (christianbiring1) [inline app.rb]: - [x] While developing software, the separation of concern is a very crucial point for it allows readability of your codebase and make it more organized. as specified in the project requirements kindly make sure the methods `operate`, `showmenu` and `begin` are inside the `main` method and the method is in a file called `main.rb`

**Review by christianbiring1 — CHANGES_REQUESTED** [review]
Hi @dev89s , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
# STATUS: REQUEST CHANGES :recycle: 


## Highlights
- [ ] Great work for adding the UI for your App as required :1st_place_medal: :100: 
- [ ] Descriptive PR `title` and `summary` :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (dev89s) [reply app.rb]: I get an error `Cyclomatic complexity for operate is too high` when I add the 7th `when` to the switch case. Also, the logic of the app ends the loop (and also the app) if `command == 7`

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @dev89s 

Your project is complete! Great job for making the changes 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request ✔️ 
- Works as expected✔️ 
- No linter errors✔️ 

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

tufoin

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**